arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/continuations/Effect.kt

```diff
@@ -709,36 +699,57 @@ internal class Token {
   "This will become private in Arrow 2.0, and is not going to be visible from binary anymore",
   level = DeprecationLevel.WARNING
 )
-internal class FoldContinuation<B>(
-  private val token: Token,
+internal class FoldContinuation<R, B>(
   override val context: CoroutineContext,
   private val error: suspend (Throwable) -> B,
-  private val parent: Continuation<B>
-) : Continuation<B> {
+  private val parent: Continuation<B>,
+) : Continuation<B>, Token(), EffectScope<R> {
```

Comment:
I think `main` was merged a couple times instead of `arrow2` branch. Not sure how do-able rebasing on `arrow2` would be, or cherry picking last couple commits.

Reply:
ohh, damn! yes! I made a mistake! merged with main instead of `arrow2` 🤦🏻

Reply:
let me try to do a rebase with `arrow2` let's see if I don' die